lib/Default/MySqlProperties.class.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types=1);
+
+use Dotenv\Dotenv;
+
+class MySqlProperties {
+	private const CONFIG_MYSQL_HOST = "MYSQL_HOST";
+	private const CONFIG_MYSQL_USER = "MYSQL_USER";
+	private const CONFIG_MYSQL_PASSWORD = "MYSQL_PASSWORD";
+	private const CONFIG_MYSQL_PORT = "MYSQL_PORT";
+	private const CONFIG_MYSQL_DATABASE = "MYSQL_DATABASE";
+	private string $host;
+	private string $user;
+	private string $password;
+	private int $port;
+	private string $databaseName;
+
+	public function __construct() {
+		$environmentFile = self::determineConfigEnvironmentFile();
+		$config = Dotenv::createArrayBacked(ROOT, $environmentFile);
+		$array = $config->load();
+		self::validateConfig($config);
+		[
+			MySqlProperties::CONFIG_MYSQL_HOST => $this->host,
+			MySqlProperties::CONFIG_MYSQL_USER => $this->user,
+			MySqlProperties::CONFIG_MYSQL_PASSWORD => $this->password,
+			MySqlProperties::CONFIG_MYSQL_PORT => $port,
+			MySqlProperties::CONFIG_MYSQL_DATABASE => $this->databaseName,
+		] = $array;
+		$this->port = (int)$port;
+	}
+
+	public static function determineConfigEnvironmentFile(): string {
```

Comment:
Are these two methods public solely for the purpose of accessing them in unit testing? If so, I still feel that the interface should reflect its intended use, but could be convinced otherwise. It looks like there is a way to unit test private methods without making them public -- what do you think about https://jtreminio.com/blog/unit-testing-tutorial-part-iii-testing-protected-private-methods-coverage-reports-and-crap/#targeting-private-protected-methods-directly?